### PR TITLE
Added AiLibrary package

### PR DESCRIPTION
**Pull request by starobinskii**
Hey guys!
I'm not sure that I am doing everything correctly, but I have created a package and now I want to add it to the Spack repository. If that's OK)

### CODE REVIEW

File: var/spack/repos/builtin/packages/ailibrary/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ailibrary(Package):
+    """AiLibrary is a single-header C++ Library from Ailurus Studio that brings you extra time to admire life instead of coding the same functions again and again."""
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ailibrary/package.py]: please make multi-line (IIRC it's 80 chars max), one can run `spack flake8` locally to check such things.

File: var/spack/repos/builtin/packages/ailibrary/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    homepage = "https://github.com/starobinskii/AiLibrary"
+    url      = "https://github.com/starobinskii/AiLibrary/archive/v1.3.0.tar.gz"
+
+    version('1.3.0', '56f59bbacfc242f9b8b38d9f2b8133ec')
+
+    sanity_check_is_file = ['include/ai.hh']
+    sanity_check_is_file = ['include/ai']
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ailibrary/package.py]: you probably can remove the second one.

File: var/spack/repos/builtin/packages/ailibrary/package.py (hunk 1 above)
Comment (starobinskii) [reply var/spack/repos/builtin/packages/ailibrary/package.py]: Sure thing

File: var/spack/repos/builtin/packages/ailibrary/package.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+class Ailibrary(Package):
+    """AiLibrary is a single-header C++ Library from Ailurus Studio
+    that brings you extra time to admire life instead of coding the
+    same functions again and again
+    """
+
+    homepage = "https://github.com/starobinskii/AiLibrary"
+    url      = "https://github.com/starobinskii/AiLibrary/archive/v1.3.0.tar.gz"
+
+    version('1.3.0', '56f59bbacfc242f9b8b38d9f2b8133ec')
```
Comment (healther) [inline var/spack/repos/builtin/packages/ailibrary/package.py]: could you use the sha256 hash?

File: var/spack/repos/builtin/packages/ailibrary/package.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    sanity_check_is_file = ['include/ai.hh']
+
+    def install(self, spec, prefix):
```
Comment (healther) [inline var/spack/repos/builtin/packages/ailibrary/package.py]: out of curiosity: why isn't this a MakefilePackage with a patch method?

File: var/spack/repos/builtin/packages/ailibrary/package.py (hunk 3 above)
Comment (starobinskii) [reply var/spack/repos/builtin/packages/ailibrary/package.py]: Done.

File: var/spack/repos/builtin/packages/ailibrary/package.py (hunk 4 above)
Comment (starobinskii) [reply var/spack/repos/builtin/packages/ailibrary/package.py]: This is my first attempt, so I tried to be as close as possible to the tutorial. But your way sounds like a more appropriate solution for me, I'll try it out)

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to address the review by @healther and get this merged.

**Pull request closed without merging**